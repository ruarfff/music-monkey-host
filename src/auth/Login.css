.Login {
  background-color: #222;
  height: 100%;
}

.Login-header {
  height: 4em;
  padding: 1em;
  text-align: center;
}

.Login-title {
  font-size: 2.5em;
  color: ivory;
}

.Login-content {
  text-align: center;
  margin-top: 4em; 
}

.Login-button {
  height: 8em;
  width: 16em;
}
