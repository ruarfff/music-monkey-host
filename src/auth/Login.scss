.Login {
  background-color: #fafafa;
  width: 420px;
  height: 620px;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.Login-logo {

}

.Login-overlay {
  background-color: rgba(81, 89, 166, 0.48);
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Login-background {
  background-position-y: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  background-image: url('../assets/club.png');
}

.Login-logo-section {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 44px;
  height: 100px;
}

.Login-text-section {
  height: 200px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  font-family: 'Roboto', 'sans-serif';
}

.Login-title {
  font-size: 30px;
  line-height: 20px;
  text-align: center;
  font-weight: 400;
  letter-spacing: 0.25px;
}

.Login-subtitle {
  font-size: 18px;
  line-height: 20px;
  text-align: center;
  letter-spacing: 0.25px;
  font-weight: 400;
}

.Login-content {
  text-align: center;
  margin-top: 4em;
}

.Login-spotify-button {
  margin-bottom: 8em;
}
