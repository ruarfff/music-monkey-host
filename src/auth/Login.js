import React, { Component } from 'react'
import PropTypes from 'prop-types'
import logo from './spotify-login.svg'
import LoadingSpinner from '../loading/LoadingSpinner'
import './Login.css'

const serviceUrl = process.env.REACT_APP_MM_API_URL
const redirectToUrl = process.env.REACT_APP_REDIRECT_URL

class Login extends Component {
  componentDidMount() {
    this.props.login()
  }

  render() {
    if (this.props.auth.isAuthenticating) {
      return <LoadingSpinner />
    }
    return (
      <div className="Login">
        <div className="Login-header">
          <h1 className="Login-title">Welcome to MusicMonkey for Hosts</h1>
        </div>
        <div className="Login-content">
          <a href={serviceUrl + '/login?redirectTo=' + redirectToUrl}>
            <img src={logo} className="Login-button" alt="login" />
          </a>
        </div>
      </div>
    )
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
}

export default Login
