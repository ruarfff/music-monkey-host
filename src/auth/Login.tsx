import * as React from 'react'
import logo from '../assets/logo-home.svg'
import IAction from '../IAction'
import LoadingSpinner from '../loading/LoadingSpinner'
import ErrorNotification from '../util/ErrorNotification'
import IAuthState from './IAuthState'
import './Login.scss'
import spotifyLoginButtonImage from './spotify-login.svg'

const serviceUrl = process.env.REACT_APP_MM_API_URL
const authSuffix = process.env.REACT_APP_AUTH_SUFFIX

interface ILoginProps {
  auth: IAuthState
  clearAuthError(): IAction
  login(): IAction
}

class Login extends React.Component<ILoginProps, {}> {
  public componentDidMount() {
    this.props.login()
  }

  public render() {
    const { isAuthenticating, authError } = this.props.auth
    if (isAuthenticating) {
      return <LoadingSpinner />
    }

    const spotifyLoginUrl = serviceUrl + '/auth/spotify-host' + authSuffix

    return (
      <div className="Login-background">
        <div className="Login-overlay">
          <div className="Login">
            <div className="Login-logo-section">
              <img className="Login-logo" src={logo} alt="logo" />
            </div>
            <div className="Login-text-section">
              <h1 className="Login-title">Playlist for Everyone</h1>
              <h2 className="Login-subtitle">
                Get started today, login or sign
                <br />
                up with your spotify account.
              </h2>
            </div>
            <div className="Login-content">
              <div>
                <a href={spotifyLoginUrl}>
                  <img
                    src={spotifyLoginButtonImage}
                    className="Login-spotify-button"
                    alt="login"
                  />
                </a>
              </div>

              {authError &&
                authError.errorContext === 'host-login' && (
                  <ErrorNotification
                    message={
                      (authError.response && authError.response.data) ||
                      authError.message
                    }
                    onClose={this.handleErrorAcknowledged}
                  />
                )}
            </div>
          </div>
        </div>
      </div>
    )
  }
  private handleErrorAcknowledged = () => {
    this.props.clearAuthError()
  }
}

export default Login
