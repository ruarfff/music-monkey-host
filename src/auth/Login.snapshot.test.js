import React from 'react'
import renderer from 'react-test-renderer'
import Login from './Login'

describe('Login Snapshot', () => {
  it('has not changed unexpectedly', () => {
    const auth = {}
    const login = () => {}
    const component = renderer.create(<Login auth={auth} login={login} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
