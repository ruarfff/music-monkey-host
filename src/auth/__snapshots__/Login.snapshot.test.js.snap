// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot has not changed unexpectedly 1`] = `
<div
  className="Login"
>
  <div
    className="Login-header"
  >
    <h1
      className="Login-title"
    >
      Welcome to MusicMonkey for Hosts
    </h1>
  </div>
  <div
    className="Login-content"
  >
    <a
      href="http://music-monkey-api.eu-west-1.elasticbeanstalk.com/login?redirectTo=http://musicmonkeyhost-hosting-mobilehub-246253618.s3-website.eu-west-1.amazonaws.com/callback"
    >
      <img
        alt="login"
        className="Login-button"
        src="spotify-login.svg"
      />
    </a>
  </div>
</div>
`;
